keanu-project/src/test/java/io/improbable/keanu/vertices/dbl/DoubleVertexTest.java

```diff
@@ -98,4 +111,41 @@ public void canCastToIntegerVertex() {
         IntegerVertex intV1 = v1.toInteger();
         assertEquals(3, intV1.getValue().scalar().longValue());
     }
+
+    @Test
+    public void canMatrixMultiplyVectors() {
+        DoubleVertex vectorsMultiplied = ConstantVertex.of(vectorA).matrixMultiply(ConstantVertex.of(vectorA));
+        DoubleTensor result = vectorsMultiplied.lazyEval();
+        DoubleTensor expectedResult = DoubleTensor.scalar(5.);
+        assertThat(result, valuesAndShapesMatch(expectedResult));
+    }
+
+    @Test
+    public void canMatrixMultiplyVectorAndMatrix() {
+        DoubleVertex vectorsMultiplied = ConstantVertex.of(vectorA).matrixMultiply(ConstantVertex.of(vectorB.reshape(2, 3)));
```

Comment:
nit: `vectorsMultiplied` is an inaccurate name

Reply:
😊